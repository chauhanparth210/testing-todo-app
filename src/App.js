import React, { useState } from "react";
import { Input } from "./components/Input";
import { Todos } from "./components/Todos";

function App() {
  const [todos, setTodos] = useState([]);

  const addTodo = (todo) => {
    setTodos([...todos, todo]);
  };

  return (
    <div className="todo-app">
      <Input addTodo={addTodo} />
      <Todos todos={todos} />
    </div>
  );
}

export default App;
