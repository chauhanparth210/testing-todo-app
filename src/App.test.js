import React from "react";
import { mount } from "enzyme";
import TestRenderer from 'react-test-renderer';
import App from "./App";

describe("App component testing", () => {
  let wrapper;
  const todoValue = "I need to poo ğŸ’©";
  beforeEach(() => {
    wrapper = mount(<App />);
    expect(wrapper.exists()).toBeTruthy();
  });

  it("should render App component properly",()=>{
    const testRenderer = TestRenderer.create(<App />);
    expect(testRenderer).toMatchSnapshot();
  })

  // it("initially input should be empty", () => {
  //   const inputElement = wrapper.find("input[type='text']");
  //   expect(inputElement.props().value).toBe("");
  // });

  // describe("it could be populated with a value", () => {
  //   beforeEach(() => {
  //     const inputElement = wrapper.find("input[type='text']");
  //     inputElement.simulate("change", { target: { value: todoValue } });
  //   });

  //   it(`the input value changes to ${todoValue}`, () => {
  //     const inputElement = wrapper.find("input[type='text']");
  //     expect(inputElement.props().value).toBe(`${todoValue}`);
  //   });
  // });

  describe("`then` the form can be submitted", () => {
    beforeEach(() => {
      const inputElement = wrapper.find("input[type='text']");
      inputElement.simulate("change", { target: { value: todoValue } });
      const form = wrapper.find(".input-form");
      form.simulate("submit", {
        preventDefault: () => {},
      });
    });

    describe("when the form is submitted", () => {
      // it("the input field is cleared", () => {
      //   const input = wrapper.find("input[type='text']");
      //   expect(input.props().value).toBe("");
      // });

      // it("the todovalue is added to the todolist", () => {
      //   const todoList = wrapper.find(".todos");
      //   expect(todoList).toHaveLength(1);
      // });

      // it(`the todovalue is  ${todoValue}`, () => {
      //   const todoList = wrapper.find(".todos");
      //   expect(todoList.text()).toBe(todoValue + "x");
      // });

      it("should render App component with one todo item",()=>{
        const testRenderer = TestRenderer.create(<App />);
        expect(testRenderer).toMatchSnapshot();
      })

      it("clicking on removeTodo button", () => {
        const removeTodoBtnElement = wrapper.find(".todos>li>button");
        removeTodoBtnElement.simulate("click");
        let todosListElmt = wrapper.find(".todos");
        expect(todosListElmt.find("ul").children()).toHaveLength(0);
      });

      it("should render App component after deleting single todo",()=>{
        const testRenderer = TestRenderer.create(<App />);
        expect(testRenderer).toMatchSnapshot();
      })
    });
  });
});
