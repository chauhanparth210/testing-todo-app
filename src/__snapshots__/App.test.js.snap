// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component testing \`then\` the form can be submitted when the form is submitted should render App component after deleting single todo 1`] = `
<div
  className="todo-application"
>
  <form
    className="input-form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      add todo
    </button>
  </form>
  <ul
    className="todos"
  />
</div>
`;

exports[`App component testing \`then\` the form can be submitted when the form is submitted should render App component with one todo item 1`] = `
<div
  className="todo-application"
>
  <form
    className="input-form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      add todo
    </button>
  </form>
  <ul
    className="todos"
  />
</div>
`;

exports[`App component testing should render App component properly 1`] = `
<div
  className="todo-application"
>
  <form
    className="input-form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      add todo
    </button>
  </form>
  <ul
    className="todos"
  />
</div>
`;
